pepys_timeline/db.py

```diff
@@ -23,30 +26,38 @@ def get_db_conn_kwargs():
     return db_params
 
 
-def get_config_options():
+def get_query_result(query, vars_=None):
     db_conn_kwargs = get_db_conn_kwargs()
-    with psycopg2.connect(**db_conn_kwargs) as conn:
-        with conn.cursor(cursor_factory=RealDictCursor) as curs:
-            curs.execute(CONFIG_OPTIONS_QUERY)
-            config_options = curs.fetchall()
-    return config_options
+    try:
+        with psycopg2.connect(**db_conn_kwargs) as conn:
+            try:
+                with conn.cursor(cursor_factory=RealDictCursor) as curs:
+                    curs.execute(query, vars_)
+                    result = curs.fetchall()
+            except psycopg2.Error:
+                raise DatabaseQueryError('Error querying database.')
+    except psycopg2.Error:
+        raise DatabaseConnectionError('Error connecting to database.')
+    return result
+
+
+def get_config_options():
+    res = get_query_result(CONFIG_OPTIONS_QUERY)
+    refresh_row = [
```

Comment:
This code looks a bit clumsy - it looks like you're editing `refresh_row` and then returning `res`, and it doesn't seem clear that editing `refresh_row` edits the value in `res`. I would edit `res` directly instead.

Reply:
@robintw good point, I refactored this.